### DISCUSSION TITLE
Supabase with PostgreSQL 16

### DISCUSSION THREAD

**Discussion by dbi1** [supabase/supabase#19427] [General]
Hello,

Thinking of using PostgreSQL 16 with Supabase. Does anybody know if there are issues in going with v16 rather than v15?

Thank you.

**Comment by AntonOfTheWoods** [comment]
So no one has tried? Or this is documented in a super obvious place maybe? Google didn't turn anything up with my search terms...

**Reply by cmoyd76** [reply]
Late to the party but I'm attempting the same thing now. Were you able to make it work?

**Reply by AntonOfTheWoods** [reply]
One of the core extensions hasn't been modified to work with 16 yet (https://github.com/supabase/supautils/issues/78, if memory serves). I tested at least updating to 15.6 by simply changing the `Dockerfile` to 

```
...
ARG postgresql_release=${postgresql_major}.6
...
```

And that seems to work flawlessly. I have no idea why they wouldn't be interested in at least updating to the latest, secure version of `15` but I am maybe missing something...

**Reply by kuncevic** [reply]
I am looking forward trying it with v16 tho, is it the right docker compose to modify https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml ? 

update: just looked into this https://discord.com/channels/839993398554656828/1224702986865999902/1224702986865999902, I suppose they must announce when v16 is officially supported, right?

**Reply by AntonOfTheWoods** [reply]
PG17 has now been released and still no word on 16, though `supautils` now claims to support 16, in spite of the above ticket not being resolved. I guess none of their customers have been requesting it!

**Reply by PylotLight** [reply]
> PG17 has now been released and still no word on 16, though `supautils` now claims to support 16, in spite of the above ticket not being resolved. I guess none of their customers have been requesting it!

Time to try a separate v17 alpine instance with sb to see if it works?

**Reply by AntonOfTheWoods** [reply]
Ok, so looks like at least this may be moving more quickly this time. https://github.com/supabase/supautils/commits/sam/pg17/ seems to be upgrading `supautils` to 17, which means we may get 17 support at the same time 16 is available, and that might be sooner rather than later!

EDIT: and maybe even full support here! https://github.com/supabase/postgres/tree/sam/postgres-17-0

EDIT2: looks like `supautils` now officially supports 17. Now we maybe just have to wait for Mr Rose to get his branch merged. I really want to update my helm chart but I'm not starting until we get 17!

**Reply by nthtrung09it** [reply]
Hi @AntonOfTheWoods does your helm chart support multiple nodes? ( does it work with k3s ? )

**Reply by AntonOfTheWoods** [reply]
> Hi @AntonOfTheWoods does your helm chart support multiple nodes? ( does it work with k3s ? )

It is based on the now-abandoned `bitnami` chart, with the missing bits added. It does a couple of slightly hacky things (like adding a second `bitnami` `postgresql` for the logging, etc.) but is definitely intended to be multi-node. I am also using `k3d`/`k3s` so it's definitely compatible with that - other stuff will also need to get tested at some point though!

That said, I spent several weeks migrating a large project to `electric-sql` before they did a hard pivot and left many of us with large turds on our faces... And I was going to migrate to a `helm` managed `supabase` when I finished that. As that didn't happen, I put the chart on the backburner and haven't touched it since June. I was waiting for `pg17` on `supabase` to pick it back up (without `electric`...). See https://github.com/supafull/helm-charts and https://github.com/supafull/helm-charts/tree/main/charts/supabase . Any help/feedback most welcome - it definitely wasn't battle-tested so is still a work in progress!

**Comment by shadoath** [comment]
Will there be an option to update my db? I'm still on 15 and it says it is the latest.
![Screenshot 2024-11-27 at 10 30 47 AM](https://github.com/user-attachments/assets/6387e6b7-a966-48a6-9245-cbc9e33f489e)

**Comment by julioxavierr** [comment]
The streets need it!!

**Comment by psirenny** [comment]
It looks like progress on upgrading Postgres has stalled, which is unnerving, considering that Postgres is now on version 17.

**Reply by AntonOfTheWoods** [reply]
They are making progress... but I think people aren't entirely clear on how they operate for the most part. Supabase is a SAAS provider of services to developers/projects. They don't make money directly from the open source part, and if their clients aren't saying "ok, well I have a budget of $1M per year to spend on a platform like yours, but because you don't support version X, I'll have to go elsewhere", there is pretty much zero reason for them to spend time (=money) on this. 

If you look at the *public* tags and branches on `supabase/postgres`, it is a hive of activity, and there was just a merge that (appears to) supports `postgres 17` with the `orioledb` patches - with all the extensions (and maybe a couple more?). But it is difficult to build and they definitely aren't prioritising this kind of documentation. The way they develop and build has been completely revamped over the last few months (again, just look at the commits) and there haven't been any documentation updates. At least not any public updates (https://github.com/supabase/postgres/tree/develop/nix/docs).

From looking at the build logs it looks like they are building on every commit, as least running their test suite, but they are only keeping the `pg15` images, at least on public servers. Again, they could have the docker images somewhere (like on github for example!) but they don't. They also don't appear to want the devs to communicate with the community... at least not on this. Maybe the devs can communicate if they want to (in their spare time for example) but they don't appear so inclined...

**Reply by AntonOfTheWoods** [reply]
https://supabase.com/blog/nix-postgres

**Comment by cstork** [comment]
Thinking of how many benefits an upgrade – especially to version 17 – would give us, this seem to be worth as much as some flashy features.  A low hanging fruit, some might say. ;-)

**Comment by ManuelAngel99** [comment]
Seems like they are working on it, there are docker images for postgres 17 in the supabase:postgres dockerhub repo:

https://hub.docker.com/layers/supabase/postgres/17.0.1.011-orioledb/images/sha256-fa91d4c31235b28c640ddea8cd3527e6027f42d222d8f15b2458e26ed5cd566e

**Reply by AntonOfTheWoods** [reply]
They've been working on it for ages but were only publishing the pg15 images in public repos. But looks like we now have publicly available images, so they may well make that available on the SAAS soon too.

**Comment by samuba** [comment]
Have to say this: 
The fact that a Postgres company is not able to provide supoprt for the last **two** Postgres major versions is pretty concerning and makes me lose trust in the capabilities of Supabase as a company. To me, keeping up with Postgres development is Table stakes for a Postgres company.
I would at least expect some official communication about it to explain the current state and a rough timeline for when these versions will be supported. Including plans/assurance on how 1-2 year version delays wont happen in the future anymore.

Neon supported Postgres 17 the same day it got released. Thats what I would expect from a company thats so Postgres centric. Just sayin.

Ive been a Supabase customer for multiple years now and I love the product. So I'm here rooting for Supabase to win the game. Sorry if this came of too negative, @kiwicopple. But I really think you guys should address this to restore trust.
